### PR TITLE
Convert Alloy Data to Rivets.Server

### PR DESCRIPTION
**Pull request by Alva8756**

### CODE REVIEW

File: internal/util/inventoryconverter.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	}
+
+	return &rivets.Server{
+		BIOSCfg:  biosCfg,
+		ID:       serverID,
+		Facility: facility,
+		// Name:
+		// BMCAddress:
+		// BMCUser:
+		// BMCPassword: device.BMC.
```
Comment (Alva8756) [inline internal/util/inventoryconverter.go:35]: Hi Vince,
I am not sure do we want to fill in BMC credentials info?
If so, do we want to let Alloy sending those info to use? Or do we query the credentials inside CIS?

File: internal/util/inventoryconverter.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+			Capabilities: devMainBoard.Capabilities,
+		},
+		devMainBoard.Status,
+		devMainBoard.Firmware,
+	)
+	if component != nil {
+		components = append(components, component)
+	}
+
+	// memory
```
Comment (Alva8756) [inline internal/util/inventoryconverter.go:126]: If I am converting the bios/bmc/mainboard correctly, I will continue fill memory/nics/drivers... component.

File: internal/util/inventoryconverter.go (hunk 1 above)
Comment (DoctorVin) [reply internal/util/inventoryconverter.go:35]: For our purposes here skip the BMC details. The Name can be set as just the ID uuid as a string. In order for CIS to have any of that stuff Alloy would have to send it along, but those details came from FleetDB in the first place. Let's just avoid it.

File: internal/util/inventoryconverter.go (hunk 2 above)
Comment (DoctorVin) [reply internal/util/inventoryconverter.go:126]: This looks good so far.

File: internal/inventoryconverter/inventoryconverter.go [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+		}
+	}
+
+	// nics
+	if devNICs := device.NICs; len(devNICs) > 0 {
+		for idx, dn := range devNICs {
+			if strings.TrimSpace(dn.Serial) == "" {
+				dn.Serial = strconv.Itoa(idx)
+			}
+			//  _, p := range dn.NICPorts
```
[the remark this answers is not in the document]
Comment (DoctorVin) [reply internal/inventoryconverter/inventoryconverter.go:177]: I don't have a great answer for you right now. For now, ignore that and concentrate on the rest of the components. We can discuss this later on Slack. We might need to extend `rivets`, but right now I don't have a great plan.

File: internal/inventoryconverter/inventoryconverter.go [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+			}
+			component := ic.newComponent(common.SlugNIC, dn.Vendor, dn.Model, dn.Serial, dn.ProductName, nil, dn.Status, dn.Firmware)
+			if component != nil {
+				components = append(components, component)
+			}
+		}
+	}
+
+	// drives
+	if devDrivers := device.Drives; len(devDrivers) > 0 {
```
Comment (DoctorVin) [inline internal/inventoryconverter/inventoryconverter.go:185]: `s/Drivers/Drives/g`

**Review by DoctorVin — CHANGES_REQUESTED** [review]
typo, but otherwise looks fine.

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**